src/pages/announcements/utils.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import yup from '../../utils/yup-config';
+import { Duration, parse, serialize } from 'tinyduration';
+
+export const ID = 'id';
+export const DATE = 'creationDate';
+export const MESSAGE = 'message';
+export const DURATION = 'duration';
+export const DAYS = 'days';
+export const HOURS = 'hours';
+export const MINUTES = 'minutes';
+
+export interface DurationFormData {
+    [DAYS]: number | null;
+    [HOURS]: number | null;
+    [MINUTES]: number | null;
+}
+
+export interface AnnouncementFormData {
```

Comment:
You can extract this type from yup schema directly

Reply:
What do you mean ?
One problem is that we use null and yup does not, so the types are not compatibles.
Maybe we should ban null as said Quentin but it's something we need to decide globally because lot of forms use our custom input components with null values.
https://github.com/gridsuite/commons-ui/blob/148631dd4688134898f6163a10a85fcc041e1730/src/components/react-hook-form/numbers/integer-input.jsx#L11